pkg/operator/app/workflowrundeps.go

```diff
@@ -177,12 +204,21 @@ func WorkflowRunDepsWithToolInjectionPool(pr pvpoolv1alpha1.PoolReference) Workf
 	}
 }
 
+func WorkflowRunDepsWithNamespace(ns *corev1obj.Namespace) WorkflowRunDepsOption {
```

Comment:
Did you end up needing this code path? If not, I'd prob remove this -- feels a bit confusing to let this be overridden.